### PR TITLE
Docs on fromSinkAndSource

### PR DESCRIPTION
**Pull request by johanandren**
Refs #25468

### CODE REVIEW

**Comment by johanandren** [comment]
I think I need an additional sample, perhaps also one for coupled, but need to come up with something reasonably realistic.

File: akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md [hunk 1]
```diff
@@ -4,19 +4,63 @@ Creates a `Flow` from a `Sink` and a `Source` where the Flow's input will be sen
...
 @ref[Flow operators composed of Sinks and Sources](../index.md#flow-operators-composed-of-sinks-and-sources)
 
-@@@div { .group-scala }
-
 ## Signature
 
-@@signature [Flow.scala](/akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala) { #fromSinkAndSource }
-
-@@@
+@apidoc[Flow.fromSinkAndSource](Flow$) { scala="#fromSinkAndSource[I,O](sink:akka.stream.Graph[akka.stream.SinkShape[I],_],source:akka.stream.Graph[akka.stream.SourceShape[O],_]):akka.stream.scaladsl.Flow[I,O,akka.NotUsed]" java="#fromSinkAndSource(akka.stream.Graph,akka.stream.Graph)" }
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md]: (tested in browser :+1: )

File: akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md [hunk 2]
```diff
@@ -4,19 +4,63 @@ Creates a `Flow` from a `Sink` and a `Source` where the Flow's input will be sen
...
 
 ## Description
 
-Creates a `Flow` from a `Sink` and a `Source` where the Flow's input will be sent to the `Sink`
-and the `Flow` 's output will come from the Source.
+<img src="../../../images/fromSinkAndSource.png" alt="Diagram" width="350"/>
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md]: Nice pictures! Not tested on mobile but looks great on desktop

File: akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md [hunk 3]
```diff
@@ -4,19 +4,63 @@ Creates a `Flow` from a `Sink` and a `Source` where the Flow's input will be sen
...
+Scala
+:   @@snip [FromSinkAndSource.scala](/akka-docs/src/test/scala/docs/stream/operators/flow/FromSinkAndSource.scala) { #chat }
+
+Java
+:   @@snip [FromSinkAndSource.java](/akka-docs/src/test/java/jdocs/stream/operators/flow/FromSinkAndSource.java) { #chat }
+
+
+The same patterns can also be applied to [Akka HTTP WebSockets](https://doc.akka.io/docs/akka-http/current/server-side/websocket-support.html#server-api) which also has an API accepting a `Flow` of messages. 
+
+`fromSinkAndSource` can also be useful when testing a component that takes a `Flow` allowing for complete separate control and assertion of incoming and outgoing elements using stream testkit test probes for sink and source:
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md]: perhaps mention here again that `Flow.fromSinkAndSourceCoupled` can be used when you want the client to be able to control when the exchange completes?

File: akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSourceCoupled.md [hunk 4]
```diff
@@ -4,20 +4,15 @@ Allows coupling termination (cancellation, completion, erroring) of Sinks and So
...
 @ref[Flow operators composed of Sinks and Sources](../index.md#flow-operators-composed-of-sinks-and-sources)
 
-@@@div { .group-scala }
-
 ## Signature
 
-@@signature [Flow.scala](/akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala) { #fromSinkAndSourceCoupled }
-
-@@@
+@apidoc[Flow.fromSinkAndSourceCoupled](Flow$) { scala="#fromSinkAndSourceCoupled[I,O](sink:akka.stream.Graph[akka.stream.SinkShape[I],_],source:akka.stream.Graph[akka.stream.SourceShape[O],_]):akka.stream.scaladsl.Flow[I,O,akka.NotUsed]" java="#fromSinkAndSourceCoupled(akka.stream.Graph,akka.stream.Graph)" }
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSourceCoupled.md]: :+1:  as well

**Review by raboof — APPROVED** [review]
Great examples and explanation

File: akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md [hunk 5]
```diff
@@ -4,19 +4,63 @@ Creates a `Flow` from a `Sink` and a `Source` where the Flow's input will be sen
...
 
-Creates a `Flow` from a `Sink` and a `Source` where the Flow's input will be sent to the `Sink`
-and the `Flow` 's output will come from the Source.
+<img src="../../../images/fromSinkAndSource.png" alt="Diagram" width="350"/>
+
+`fromSinkAndSource` combines a separate `Sink` and `Source` into a `Flow`.
+
+Useful in many cases where an API requires a `Flow` but you want to provide a `Sink` and `Source` whose flows of elements are decoupled.
+
+Note that termination events, like completion and cancelation is not automatically propagated through to the "other-side" of the such-composed Flow. The `Source` can complete and the sink will continue to accept elements.
```
Comment (ennru) [inline akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md]:
```suggestion
Note that termination events, like completion and cancellation, are not automatically propagated through to the "other-side" of the such-composed `Flow`. The `Source` can complete and the sink will continue to accept elements.
```

File: akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md [hunk 6]
```diff
@@ -4,19 +4,63 @@ Creates a `Flow` from a `Sink` and a `Source` where the Flow's input will be sen
...
+
+Use @ref:[fromSinkAndSourceCoupled](fromSinkAndSourceCoupled.md) if you want to couple termination of both of the ends. 
+
+## Examples
+
+One use case is for constructing a TCP server which where request and response does not map 1:1 (like it does in the @ref[Echo TCP server sample](../../stream-io.md) where every incoming test is echoed back) but allows separete flows of elements from the client to the server and from the server to the client.
```
Comment (ennru) [inline akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md]:
```suggestion
One use case is constructing a TCP server where requests and responses do not map 1:1 (like it does in the @ref[Echo TCP server sample](../../stream-io.md) where every incoming test is echoed back) but allows separate flows of elements from the client to the server and from the server to the client.
```

File: akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md [hunk 7]
```diff
@@ -4,19 +4,63 @@ Creates a `Flow` from a `Sink` and a `Source` where the Flow's input will be sen
...
+
+Scala
+:   @@snip [FromSinkAndSource.scala](/akka-docs/src/test/scala/docs/stream/operators/flow/FromSinkAndSource.scala) { #halfClosedTcpServer }
+
+Java
+:   @@snip [FromSinkAndSource.java](/akka-docs/src/test/java/jdocs/stream/operators/flow/FromSinkAndSource.java) { #halfClosedTcpServer }
+
+With this server running you could use `telnet 127.0.0.1 9999` to see a stream of timestamps being printed, one every second. 
+
+The following sample is a little bit more advanced and uses the `MergeHub` to dynamically merge incoming messages to a single stream which is then fed into a `BroadcastHug` which emits elements over a dynamic set of downstreams allowing us to create a simplistic little TCP chat server in which a text entered from one client is emitted to all connected clients.
```
Comment (ennru) [inline akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md]:
```suggestion
The following sample is a little bit more advanced and uses the @apidoc[MergeHub] to dynamically merge incoming messages to a single stream which is then fed into a @apidoc[BroadcastHub] which emits elements over a dynamic set of downstreams allowing us to create a simplistic little TCP chat server in which a text entered from one client is emitted to all connected clients.
```

File: akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md [hunk 8]
```diff
@@ -4,19 +4,63 @@ Creates a `Flow` from a `Sink` and a `Source` where the Flow's input will be sen
...
-of the such-composed Flow. Use `Flow.fromSinkAndSourceCoupled` if you want to couple termination of both of the ends,
-for example most useful in handling websocket connections.
+Scala
+:   @@snip [FromSinkAndSource.scala](/akka-docs/src/test/scala/docs/stream/operators/flow/FromSinkAndSource.scala) { #chat }
+
+Java
+:   @@snip [FromSinkAndSource.java](/akka-docs/src/test/java/jdocs/stream/operators/flow/FromSinkAndSource.java) { #chat }
+
+
+The same patterns can also be applied to [Akka HTTP WebSockets](https://doc.akka.io/docs/akka-http/current/server-side/websocket-support.html#server-api) which also has an API accepting a `Flow` of messages.
```
Comment (ennru) [inline akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md]:
```suggestion
The same patterns can also be applied to @extref:[Akka HTTP WebSockets](akka-http:/server-side/websocket-support.html#server-api) which also have an API accepting a `Flow` of messages. 
```

**Review by ennru — APPROVED** [review]
LGTM.

File: akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md (hunk 8 above)
Comment (johanandren) [reply akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md]: Thx, except that it doesn't work ;)

Comment (johanandren) [reply akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md]: Ah, simple fix `akka.http` not `akka-http`

Comment (ennru) [reply akka-docs/src/main/paradox/stream/operators/Flow/fromSinkAndSource.md]: Oh, sorry. That is non-standard to what we use in other repos...

**Pull request merged**